automates/text_reading/src/main/scala/org/clulab/aske/automates/apps/ExtractAndAlign.scala

```diff
@@ -218,25 +235,42 @@ object ExtractAndAlign {
     } else outputJson
   }
 
-  def getGlobalVars(descrMentions: Seq[Mention]): Seq[GlobalVariable] = {
+  def getGlobalVars(descrMentions: Seq[Mention], wikigroundings: Option[Map[String, Seq[sparqlWikiResult]]], groundToWiki: Boolean): Seq[GlobalVariable] = {
 
     val groupedVars = descrMentions.groupBy(_.arguments("variable").head.text.replace(".", ""))
     val allGlobalVars = new ArrayBuffer[GlobalVariable]()
     for (gr <- groupedVars) {
       val glVarID = randomUUID().toString()
       val identifier = gr._1//AlignmentBaseline.replaceWordWithGreek(gr._1, AlignmentBaseline.word2greekDict.toMap); for now, don't convert: text vars are already symbols and comments shouldnt be converted except for during alignment
+      val identifierComponents = if (identifier.contains("_")) {
+        if (!identifier.split("_").map(_.length).contains(1)) {
```

Comment:
@maxaalexeeva splitting strings on special characters is an expensive command to run more times than necessary (especially since this will be done for many identifiers). Can you assign the result of `identifier.split("_")` to a value and then run multiple checks on that value?

Reply:
Great catch! Thanks!